alphagov/paas-cf
# [#160197759] Separate cloud config generation from cf manifest

What
----

The cloud-config is shared by all deployments using the bosh director,
so it shouldn't be coupled to the cf manifest.

Separating it out makes it easier to see what it's dependencies are. It also means that we can re-enable the prometheus manifest tests that test against it.

As part of this I've included a few other refactorings:

* Move the test helpers relating to fixtures out into the shared space (where the fixtures themselves live)
* Remove the concourse tfstate from the manifest generation scripts as nothing was being consumed from it.
* Remove the concourse manifest from the destroy pipeline as it wasn't actually being used

How to review
-------------

* Code review
* Run the pipeline from this branch and verify that it's a no-op
* Run the init job from the destroy pipeline twice to verify the self-update-pipelines task still works (no need to run the full pipeline - the delete-deployment job can be paused).

Who can review
--------------

Not me.
